pkg/kallsyms/kallsyms.go

```diff
@@ -149,12 +151,53 @@ var (
 //
 //	kallsyms.SpecUpdateAddresses(spec, []string{"socket_file_ops"})
 func SpecUpdateAddresses(spec *ebpf.CollectionSpec, symbols []string) error {
```

Comment:
Move the `len(symbols) == 0` check on L204 here to avoid allocating things if nothing has to be done?